PTGLgraphComputation/src/plcc/Main.java

```diff
@@ -5451,7 +5510,140 @@ private static long calculateSkipNeighborNum(Molecule mol1, Molecule mol2, int m
+                                    Integer LL = 11;        // 11 = # of ligand-ligand contacts         (ligand - ligand only)
+                                    */
+
+
+                                    Integer numTotalLigContactsPair = 0;
+                                    Integer numTotalRnaContactsPair = 0;
+
+
+
+                                    Integer[] minContactDistances = new Integer[numPairContacts.length];
+                                    // Holds the minimal distances of contacts of the appropriate type (see numPairContacts, index 0 is unused)
+
+                                    Integer[] contactAtomNumInResidueA = new Integer[numPairContacts.length];
+                                    // Holds the number Atom x has in its residue a for the contact with minimal distance of that type.
+                                    // See minContactDistances and numPairContacts; index 0 is unused; index 5 + 6 are also unused (atom is obvious + always the same)
+
+                                    Integer[] contactAtomNumInResidueB = new Integer[numPairContacts.length];
+                                    // Holds the number Atom y has in its residue b for the contact with minimal distance of that type.
+                                    // See minContactDistances and numPairContacts; index 0 is unused; index 5 + 6 are also unused (atom is obvious + always the same)
+
+
+                                    for(Integer r = 0; r < numPairContacts.length; r++) {      // init all arrays
+                                        numPairContacts[r] = 0;
+                                        minContactDistances[r] = -1;    // We are looking for the smallest distance >= 0 in this function so do NOT set '0' as the initial value or everything will get fucked up!
+                                        contactAtomNumInResidueA[r] = -1;   // We HAVE to assign '-1', NOT any other value here! See comments below for details.
+                                        contactAtomNumInResidueB[r] = -1;   // We HAVE to assign '-1', NOT any other value here! See comments below for details.
+                                        // The initial value of '-1' is required for the atom index arrays because our index
+                                        // starts at '0', but geom_neo treads a '0' in a line of <pdbid>.geo as 'no contact' because
+                                        // it starts its index at '1'.
+                                        // This problem is solved by the functions in MolContactInfo: they return (our_index + 1). This
+                                        // means that:
+                                        //   1) If no contact was detected, our_index is -1 and they return 0, which means 'no contact' to geom_neo.
+                                        //   2) If a contact was detected, our_index is converted to the geom_neo index. :)
+                                    }
```

Comment:
This looks cumbersome to do just to create a new mci. I noticed that we have this block several times in our code. Ideally, the creation rules would be inside an own class or at least in MolContactInfo. Also so that the comments / documentation is not spread over several places.
How about we create a static function in MolContactInfo that returns clean arrays for numPairContacts, minContactDistances, contactAtomNumInResidueA and contactAtomNumInResidueB following the here implemented rules. That way we can have the documentation just there and we call it from here and the other code duplicates. What do you think?